tests/Tokenizer/Analyzer/AttributeAnalyzerTest.php

```diff
@@ -0,0 +1,109 @@
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Tokenizer\Analyzer;
+
+use PhpCsFixer\Tests\TestCase;
+use PhpCsFixer\Tokenizer\Analyzer\AttributeAnalyzer;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Tokenizer\Analyzer\AttributeAnalyzer
+ */
+final class AttributeAnalyzerTest extends TestCase
+{
+    /**
+     * Check it's not crashing for PHP lower than 8.0 as other tests run for PHP 8.0 only.
+     */
+    public function testNotAnAttribute(): void
+    {
+        $tokens = Tokens::fromCode('<?php class Foo { private $bar; }');
+        foreach ($tokens as $index => $token) {
+            static::assertFalse(AttributeAnalyzer::isAttribute($tokens, $index));
+        }
+    }
+
+    /**
+     * @requires     PHP 8.0
+     * @dataProvider provideIsAttributeCases
+     */
+    public function testIsAttribute(bool $isInAttribute, string $code): void
+    {
+        $tokens = Tokens::fromCode($code);
+
+        foreach ($tokens as $index => $token) {
+            if ($token->equals([T_STRING, 'Foo'])) {
+                if (isset($testedIndex)) {
+                    static::fail('Test is run against index of "Foo", multiple occurrences found.');
+                }
+                $testedIndex = $index;
+            }
+        }
+        if (!isset($testedIndex)) {
+            static::fail('Test is run against index of "Foo", not found in the code sample.');
+        }
+
+        static::assertSame($isInAttribute, AttributeAnalyzer::isAttribute($tokens, $testedIndex));
+    }
+
+    /**
+     * Test case requires to having "Foo" as it will be searched for to test its index.
+     */
+    public static function provideIsAttributeCases(): iterable
+    {
+        // Foo is not an attribute
+        foreach ([
+            '<?php Foo; #[Attr] class Bar {}',
+            '<?php Foo(); #[Attr] class Bar {};',
+            '<?php \Foo(); #[Attr] class Bar {}',
+            '<?php #[Attr] class Foo {};',
+            '<?php #[Attr] class Bar {}; Foo;',
+            '<?php #[Attr] class Bar {}; \Foo();',
+            '<?php #[Attr] class Bar { const Foo = 42; };',
+            '<?php #[Attr] class Bar { function Foo() {} };',
+            '<?php #[Attr(Foo)] class Bar {}',
+            '<?php #[Attr(Foo::Bar)] class Baz {}',
+            '<?php #[Attr(Bar::Foo)] class Baz {}',
+            '<?php #[Attr(1, 2, Foo)] class Bar {}',
+            '<?php #[Attr(Foo, 1, 2)] class Bar {}',
+            '<?php #[Attr(1, 2, Foo, 3, 4)] class Bar {}',
+            '<?php #[Attr(2 * (3 + 7), 2, Foo)] class Bar {}',
+            '<?php #[Attr(2 * (3 + 7), 2, Foo, 5)] class Bar {}',
+            '<?php #[Attr(2 * (3 + 7), 2, Foo, (16 + 4) / 5 )] class Bar {}',
+            '<?php #[Attr] function Foo() {};',
+        ] as $code) {
+            yield [false, $code];
+        }
+
+        // Foo is an attribute
+        foreach ([
```

Comment:
Should we include an explicit test for a string containing `'('` inside an attribute parameter, eg `<?php #[Foo('(')] class Bar {}` since that's how we're checking for all attributes inside of `#[]`?